### PR TITLE
client/http: add scheduleUpdateMembersInfo to get the latest members ASAP

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #7300, #7673.

### What is changed and how does it work?

```commit-message
Add `scheduleUpdateMembersInfo` to get the latest members ASAP.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/http/client.go [hunk 1]
```diff
@@ -268,6 +286,7 @@ func (ci *clientInner) membersInfoUpdater(ctx context.Context) {
 			log.Info("[pd] http client member info updater stopped", zap.String("source", ci.source))
 			return
 		case <-ticker.C:
+		case <-ci.updateMembersInfoNotifier:
```
Comment (disksing) [inline client/http/client.go:289]: fallthrough

Comment (JmPotato) [reply client/http/client.go:289]: Fixed.

File: client/http/client.go [hunk 2]
```diff
@@ -169,11 +185,13 @@ func (ci *clientInner) requestWithRetry(
...
 		err = ci.doRequest(ctx, addr, reqInfo, headerOpts...)
 		if err == nil {
-			break
+			return nil
 		}
 		log.Debug("[pd] request follower addr failed",
 			zap.String("source", ci.source), zap.Int("idx", idx), zap.String("addr", addr), zap.Error(err))
 	}
+	// Schedule the members info update if all the requests failed to get the latest members as soon as possible.
+	ci.scheduleUpdateMembersInfo()
```
Comment (HuSharp) [inline client/http/client.go:194]: better using retry mechanism

Comment (CabinfeverB) [reply client/http/client.go:194]: After using ServiceDiscovery, we can use `ScheduleCheckMemberChanged`. How about waiting for #7668

**Pull request closed without merging**